### PR TITLE
Properly shutdown SQLMetadataSegmentManager and SQLMetadataRuleManager

### PR DESCRIPTION
**Pull request by drcrallen**
In some unit tests I was doing rapid starting and stopping of the SQLMetadataSegmentManager

It was logging errors which did not need to be logged because the tasks were trying to resubmit themselves after the shutdown and failing (as they should). This is not technically an error though, and this PR uses futures to allow killing of the task before shutdown to help minimize false-errors in logs.

### CODE REVIEW

**Comment by xvrl** [comment]
@drcrallen should we update the rulemanager with the same fix as well?

File: server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java [hunk 1]
```diff
@@ -168,21 +170,26 @@ public void start()
...
           {
             @Override
             public void run()
             {
-              poll();
+              try {
+                poll();
+              }
+              catch (Exception e) {
+                log.error(e, "uncaught exception in rule manager polling thread");
```
Comment (cheddar) [inline server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java]: Should this emit an alert?

Comment (xvrl) [inline server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java]: I believe this is just replicating the existing behavior, which was handled within `ScheduledExecutors.scheduleWithFixedDelay`

Comment (xvrl) [inline server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java]: nothing against having an alert though

Comment (drcrallen) [inline server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java]: I don't think there should be an alert here without a retry strategy. An acute network hiccup should not cause an alert if it is simply recoverable.

Comment (drcrallen) [inline server/src/main/java/io/druid/metadata/SQLMetadataRuleManager.java]: I filed https://github.com/druid-io/druid/issues/1380 to track the request

File: server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java [hunk 2]
```diff
@@ -103,23 +106,27 @@ public void start()
...
           {
             @Override
             public void run()
             {
-              poll();
+              try {
+                poll();
+              }
+              catch (Exception e) {
+                log.error(e, "uncaught exception in segment manager polling thread");
```
Comment (xvrl) [inline server/src/main/java/io/druid/metadata/SQLMetadataSegmentManager.java]: I just realized this error will never get logged, `SQLMetadataSegmentManager.poll()` already logs and swallows all exceptions

**Pull request merged**